pkg/engine/planner/logical/format_tree_test.go

```diff
@@ -19,200 +19,75 @@ func (t *testDataSource) Name() string          { return t.name }
-		name: "users",
-		schema: schema.Schema{
-			Columns: []schema.ColumnSchema{
-				{Name: "id", Type: schema.ValueTypeUint64},
-				{Name: "name", Type: schema.ValueTypeString},
-				{Name: "age", Type: schema.ValueTypeUint64},
-			},
-		},
-	}
-
-	scan := NewScan(ds.Name(), ds.Schema())
-	filter := NewFilter(scan, Gt("age_gt_21", Col("age"), LitI64(21)))
-	proj := NewProjection(filter, []Expr{Col("id"), Col("name")})
-
-	var f TreeFormatter
-
-	actual := "\n" + f.Format(proj)
-	t.Logf("Actual output:\n%s", actual)
-
-	expected := `
-Projection id=VALUE_TYPE_UINT64 name=VALUE_TYPE_STRING
-│   ├── Column #id
-│   └── Column #name
-└── Filter expr=age_gt_21
-    │   └── BinaryOp type=cmp op=">" name=age_gt_21
-    │       ├── Column #age
-    │       └── Literal value=21 type=VALUE_TYPE_INT64
-    └── MakeTable name=users
-`
-
-	require.Equal(t, expected, actual)
-}
-
-func TestFormatDataFrameQuery(t *testing.T) {
-	// Calculate the sum of sales per region for the year 2020
-	ds := &testDataSource{
-		name: "orders",
-		schema: schema.Schema{
-			Columns: []schema.ColumnSchema{
-				{Name: "region", Type: schema.ValueTypeString},
-				{Name: "sales", Type: schema.ValueTypeUint64},
-				{Name: "year", Type: schema.ValueTypeUint64},
-			},
-		},
-	}
-
 	df := NewDataFrame(
-		NewScan(ds.Name(), ds.Schema()),
-	).Filter(
-		Eq("year_2020", Col("year"), LitI64(2020)),
-	).Project(
-		[]Expr{
-			Col("region"),
-			Col("sales"),
-			Col("year"),
+		&MakeTable{
+			Selector: &BinOp{
+				Left:  &ColumnRef{Column: "table", Type: ColumnTypeLabel},
+				Right: LiteralString("users"),
+				Op:    BinOpKindEq,
+			},
 		},
-	).Aggregate(
-		[]Expr{Col("region")},
-		[]AggregateExpr{
-			Sum("total_sales", Col("sales")),
+	).Select(
+		&BinOp{
+			Left:  &ColumnRef{Column: "age", Type: ColumnTypeMetadata},
+			Right: LiteralInt64(21),
+			Op:    BinOpKindGt,
 		},
-	).Limit(
-		0,
-		10,
 	)
 
 	var f TreeFormatter
 
-	actual := "\n" + f.Format(df.LogicalPlan())
+	actual := "\n" + f.Format(df.Value())
 	t.Logf("Actual output:\n%s", actual)
 
 	expected := `
-Limit offset=0 fetch=10
-└── Aggregate groupings=([region]) aggregates=([total_sales])
-    │   ├── GroupExpr
-    │   │   └── Column #region
-    │   └── AggregateExpr
-    │       └── Aggregate op=sum
-    │           └── Column #sales
-    └── Projection region=VALUE_TYPE_STRING sales=VALUE_TYPE_UINT64 year=VALUE_TYPE_UINT64
-        │   ├── Column #region
-        │   ├── Column #sales
-        │   └── Column #year
-        └── Filter expr=year_2020
-            │   └── BinaryOp type=cmp op="==" name=year_2020
-            │       ├── Column #year
-            │       └── Literal value=2020 type=VALUE_TYPE_INT64
-            └── MakeTable name=orders
+Select
+│   └── BinOp op=GT
+│       ├── ColumnRef #metadata.age
+│       └── Literal value=21 kind=int64
+└── MakeTable
+        └── BinOp op=EQ
+    │       ├── ColumnRef #label.table
+    │       └── Literal value="users" kind=string
```

Comment:
This seems like a bug in the tree printer @chaudum, what do you think?

Reply:
Yes, that looks like a bug in the printer

Reply:
Fixed, though I don't know if my fix is the right one